syntax = "proto3";

package zen.remote;

import "common.proto";

service VirtualObjectService
{
  rpc New(NewResourceRequest) returns (EmptyResponse);

  rpc Delete(DeleteResourceRequest) returns (EmptyResponse);

  rpc Commit(VirtualObjectCommitRequest) returns (EmptyResponse);

  rpc Move(VirtualObjectMoveRequest) returns (EmptyResponse);

  rpc ChangeVisibility(VirtualObjectChangeVisibility) returns (EmptyResponse);
}

message VirtualObjectCommitRequest
{
  uint64 id = 1;
}

message VirtualObjectMoveRequest
{
  uint64 id = 1;
  bytes position = 2;    // vec3
  bytes quaternion = 3;  // {x y z w}
}

message VirtualObjectChangeVisibility
{
  uint64 id = 1;
  bool visible = 2;
}
